site/en/docs/extensions/mv3/service-workers/events/index.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: 'layouts/doc-post.njk'
+title: Events in service workers
+description: Extension service workers respond to both standard service worker events and many events in the extension APIs.
+date: 2023-04-19
+---
+
+Extension service workers support both [standard service worker](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope#events) events and many events in the [extension APIs](/docs/extensions/reference/). This section describes what's available and provides tips for using them.
+
+## Declaring extension events
+
+Event handlers in service workers need to be declared in the global scope, meaning they should be at the top level of the script and not be nested inside functions. This ensures that they are registered synchronously on initial script execution, which enables Chrome to dispatch events to the service worker as soon as it starts. For example:
+
+{% Compare 'worse', 'Not advised' %}
+```js/2
+chrome.storage.local.get(["badgeText"], ({ badgeText }) => {
+  chrome.browserAction.setBadgeText({ text: badgeText });
+  chrome.browserAction.onClicked.addListener(handleActionClick);
+});
+
+```
+{% endCompare %}
+
+
+{% Compare 'better', 'Better' %}
+```js/0
+chrome.action.onClicked.addListener(handleActionClick);
+
+chrome.storage.local.get(["badgeText"], ({ badgeText }) => {
+  chrome.action.setBadgeText({ text: badgeText });
+});
+```
+{% endCompare %}
+
+## Available events
```

Comment:
I think we are missing a few important Chrome API events like alarms, tabs, bookmarks, contextMenus, etc

@sebastianbenz how exhaustive did you want this list of events to be?

Reply:
I think we have two options:

1. Provide an example list of most used APIs (but make it explicit that this is only a subset).
2. Make sure to list all available examples.

2) Would be great imo, but the risk is that this gets stale over time.

Reply:
I changed this section to 'Common events' and adjusted the intro text to match. I did not add anything about seeing specific APIs for support information because I'm not confident that such information is as clear as it should be.